### PR TITLE
Add Tapbranch and Taptree constructors

### PR DESCRIPTION
**Pull request by jnewbery**
Makes things a bit easier for students by adding constructors for the Tapbranch and Taptree classes.

Also a few other small tidy-ups. Moved tagged_hash() definition out of the notebooks and renamed `Node` to `Tapbranch`

### CODE REVIEW

**Comment by jnewbery** [comment]
@jachiang - do you mind reviewing this?

File: 2.4-taptree.ipynb [hunk 1]
```diff
@@ -291,8 +279,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "taptree3 = TapTree()\n",
-    "taptree3.key = pubkey_internal\n",
+    "taptree3 = TapTree(key=pubkey_internal)\n",
```
Comment (jachiang) [inline 2.4-taptree.ipynb]: Shall we mention that the huffman constructor only constructs the root? The internal_key still needs to be set by the user.

File: test_framework/script.py [hunk 2]
```diff
@@ -1097,9 +1097,10 @@ def generate_threshold_csa(k, pubkeys):
         return tapscripts
 
 class TapTree:
-    def __init__(self):
-        self.root = Node()
-        self.key = ECPubKey()
+    def __init__(self, *, root=None, key=None):
+        """Taptree constructor. Takes an optional `root` Tapbranch and `key` ECPubKey."""
```
Comment (jachiang) [inline test_framework/script.py]: nit: Could also be a tapscript object as in chapter 2.3.

Comment (jachiang) [reply test_framework/script.py]: nit: Swapping order of args to "key, root " would be consistent with taptree descriptor order (tp(key, tree))

Comment (jnewbery) [reply test_framework/script.py]: thanks. Good suggestions.

I'll swap the order of the args so the definition here matches the descriptor format. Note however that the special `*` argument means that any arguments after need to be passed in as named args, so order doesn't matter for the callers.

File: 2.3-tapscript.ipynb [hunk 3]
```diff
@@ -518,9 +512,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "taptree = TapTree()\n",
-    "taptree.key = pubkey_internal\n",
-    "taptree.root = csahasholder_tapscript\n",
+    "taptree = TapTree(root=csahasholder_tapscript, key=pubkey_internal)\n",
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply 2.3-tapscript.ipynb]: done

File: 2.4-taptree.ipynb (hunk 1 above)
Comment (jnewbery) [reply 2.4-taptree.ipynb]: done

**Comment by jachiang** [comment]
tACK 16f84cf049d7f92bff03f08db5ef87aab6cb6a6 

Reviewed & Tested. Thanks @jnewbery!

**Comment by jachiang** [comment]
> by convention, Python uses snake_case for function and variable names, and CamelCase for class names 

Thanks, I will apply this convention in the future.

**Pull request merged**